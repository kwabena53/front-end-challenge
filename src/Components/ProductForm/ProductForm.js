import React, { useState } from "react";

import "tabler-react/dist/Tabler.css";
import { Grid, Form, Card, Button } from "tabler-react";

const ProductForm = ({ title, _productName, _price }) => {
  const productName = _productName ? _productName : undefined;
  const price = _price ? _price : undefined;
  const buttonText = _productName ? "save" : "submit";

  const [inputText, setInputText] = useState({
    prodName: productName,
    prodPrice: price,
  });

  const handleInputTextChange = (e) => {
    const target = e.target;
    setInputText({
      ...inputText,
      [target.name]: target.value,
    });
  };
  return (
    <Grid.Col lg={4} md={4} sm={12}>
      <Card styles={{ width: "300px" }}>
        <Card.Header>
          <Card.Title>{title}</Card.Title>
        </Card.Header>
        <Card.Body>
          <Form.Group label="Product Name">
            <Form.Input
              name="prodName"
              value={inputText.prodName}
              onChange={handleInputTextChange}
              placeholder="Enter product name"
            />
          </Form.Group>
          <Form.Group label="Price">
            <Form.Input
              name="prodPrice"
              value={inputText.prodPrice}
              onChange={handleInputTextChange}
              placeholder="Enter product latest price"
            />
          </Form.Group>
          <Button color="primary">{buttonText}</Button>
        </Card.Body>
      </Card>
    </Grid.Col>
  );
};

export default ProductForm;
