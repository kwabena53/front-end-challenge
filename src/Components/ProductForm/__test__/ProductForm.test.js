import { render, screen } from "@testing-library/react";
import ProductForm from "../ProductForm";
import renderer from "react-test-renderer";

test("renders product form: add product", () => {
  render(<ProductForm title="add product" />);
  const textElement = screen.getByText(/add product/i);
  expect(textElement).toBeInTheDocument();
});

test("matches product form snapshot: add product", () => {
  const component = renderer.create(<ProductForm title="add product" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders product form: edit product", () => {
  render(
    <ProductForm
      title="edit product"
      _productName="Product Name"
      _price="Price"
    />
  );
  const textElement = screen.getByText(/Product Name/i);
  expect(textElement).toBeInTheDocument();
});

test("matches product form snapshot", () => {
  const component = renderer.create(
    <ProductForm
      title="edit product"
      _productName="Product Name"
      _price="Price"
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("changes the button to save when it's an edit form", () => {
  render(
    <ProductForm
      title="edit product"
      _productName="Product Name"
      _price="Price"
    />
  );
  const textElement = screen.getByText(/save/i);
  expect(textElement).toBeInTheDocument();
});
