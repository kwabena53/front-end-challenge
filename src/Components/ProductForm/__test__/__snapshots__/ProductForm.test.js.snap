// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches product form snapshot 1`] = `
<div
  className="col col-sm-12 col-md-4 col-lg-4"
>
  <div
    className="card"
  >
    <div
      className="card-header"
      style={null}
    >
      <h3
        className="card-title"
      >
        edit product
      </h3>
    </div>
    <div
      className="card-body"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Product Name
        </label>
        <input
          className="form-control"
          name="prodName"
          onChange={[Function]}
          placeholder="Enter product name"
          type="text"
          value="Product Name"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Price
        </label>
        <input
          className="form-control"
          name="prodPrice"
          onChange={[Function]}
          placeholder="Enter product latest price"
          type="text"
          value="Price"
        />
      </div>
      <button
        className="btn btn-primary"
      >
        save
      </button>
    </div>
  </div>
</div>
`;

exports[`matches product form snapshot: add product 1`] = `
<div
  className="col col-sm-12 col-md-4 col-lg-4"
>
  <div
    className="card"
  >
    <div
      className="card-header"
      style={null}
    >
      <h3
        className="card-title"
      >
        add product
      </h3>
    </div>
    <div
      className="card-body"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Product Name
        </label>
        <input
          className="form-control"
          name="prodName"
          onChange={[Function]}
          placeholder="Enter product name"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Price
        </label>
        <input
          className="form-control"
          name="prodPrice"
          onChange={[Function]}
          placeholder="Enter product latest price"
          type="text"
        />
      </div>
      <button
        className="btn btn-primary"
      >
        submit
      </button>
    </div>
  </div>
</div>
`;
